src/auth/types/user.go

```diff
@@ -0,0 +1,51 @@
+package types
+
+import (
+	"github.com/consensys/quorum-key-manager/pkg/errors"
+	manifest "github.com/consensys/quorum-key-manager/src/manifests/types"
+)
+
+// UserInfo are extracted from request credentials by authentication middleware
+type UserInfo struct {
+	// AuthMode records the mode that succeeded to Authenticate the request ('tls', 'api-key', 'oidc' or '')
+	AuthMode string
+
+	// Tenant belonged by the user
+	Tenant string
+
+	// Username identifies the user
+	Username string
+
+	// Roles indicates the user's membership
+	Roles []string
```

Comment:
Do we need Roles here? Can't we translate the role into a list of permissions when creating the UserInfo and delete this field from the struct?

Reply:
Roles are converted into Permissions on the authorization middlewares as you mention BUT I think we will need Roles in the UserInfo because that information will be taken by the Audit Component to write records of which user with which roles accessed the app etc.